### PR TITLE
Packages API speed improvements

### PR DESCRIPTION
**Pull request by ChrisTerBeke**
Goes with https://github.com/Ultimaker/Cura/pull/6615.

It would be great to have this in 4.4, but since we're in feature freeze I can understand why you'd want this on master instead.

### CODE REVIEW

File: UM/PackageManager.py [hunk 1]
```diff
@@ -291,6 +296,24 @@ def getAllInstalledPackageIDs(self) -> Set[str]:
 
         return all_installed_ids
 
+    ## Get a list of tuples that contain the package ID and version.
+    #  Used by the Marketplace to check which packages have updates available.
+    def getAllInstalledPackageIdsAndVersions(self) -> List[Tuple[str, str]]:
+        package_ids_and_versions = []  # type: List[Tuple[str, str]]
+        all_installed_ids = self.getAllInstalledPackageIDs()
+        for package_id in all_installed_ids:
+            if package_id in self._bundled_package_dict:
```
Comment (nallath) [inline UM/PackageManager.py]: A bundled package can be updated, so this check is not correct.

File: UM/PackageManager.py [hunk 2]
```diff
@@ -291,6 +296,24 @@ def getAllInstalledPackageIDs(self) -> Set[str]:
...
+                continue
+            if package_id in self._application.getRequiredPlugins():
```
Comment (nallath) [inline UM/PackageManager.py]: A required plugin is bundled by definition. So this check will never do anything

File: UM/PackageManager.py (hunk 1 above)
Comment (ChrisTerBeke) [reply UM/PackageManager.py]: I'll remove this check then.

File: UM/PackageManager.py (hunk 2 above)
Comment (ChrisTerBeke) [reply UM/PackageManager.py]: I'll remove this check then.

**Pull request merged**